sumpy/expansion/local.py

```diff
@@ -174,8 +195,110 @@ def evaluate(self, kernel, coeffs, bvec, rscale, sac=None):
 
         return kernel.postprocess_at_target(result, bvec)
 
+    def m2l_global_precompute_nexpr(self, src_expansion):
+        """Returns number of expressions in M2L global precomputation step.
+        """
+        return len(self._m2l_global_precompute_mis(src_expansion)[1])
+
+    def _m2l_global_precompute_mis(self, src_expansion):
+        """A helper method to calculate multi-indices used in M2L global
+        precomputation step.
+        """
+        from pytools import generate_nonnegative_integer_tuples_below as gnitb
+        from sumpy.tools import add_mi
+
+        # max_mi is the multi-index which is the sum of the
```

Comment:
First things first, *what is this trying to precompute*? (It's OK for this to be a reference to the place where the overall thing is described, but without this, it's hard for me to even start understanding this.)